Section 1: Introduction and Setup/4-RoR Development Environment/testing_project/app/javascript/channels/consumer.js

```diff
@@ -0,0 +1,6 @@
+// Action Cable provides the framework to deal with WebSockets in Rails.
```

Comment:
what is this file for?

Reply:
'consumer' is the client of a WebSocket connection and consumer is created by the client-side JavaScript framework. In rails, the 'consumer.js' file is created on default to create an instance of the WebSocket connection.

Reply:
what is `client`?
What if you don't have your frontend on rails but on others. How do you magically connect to this?